error when Logic table have the same name as Actual table .

### update
it looks like a driver compatibility problem with `mysql-connector-java` of 6.0.6 or above
the issue title should updated in some days
---
i just use database sharding not table sharding, but when Logic table have the same name as Actual table, some error occurs
``` 
Caused by: java.sql.SQLException: Column name pattern can not be NULL or empty.
```
here is my config of yaml
```yaml
shardingRule:
  tables:
   # same name as Actual table， but when i use ms_ds${1..2}.t_user1, it will be ok
   # but when i change table name in mysql to t_user1，then ms_ds${1..2}.t_user1 does not work either
    t_user:
      actualDataNodes: ms_ds${1..2}.t_user
      databaseStrategy:
        inline:
          shardingColumn: id
          algorithmExpression: ms_ds${id % 2 + 1}

      keyGeneratorColumnName: id
```

i think it may be a bug

@terrymanu 
all my config, my  version is `3.1.0`
```yaml
dataSources:
  ds1: !!com.alibaba.druid.pool.DruidDataSource
    driverClassName: com.mysql.jdbc.Driver
    url: jdbc:mysql://192.168.8.129:13306/shop
    username: root
    password: root

  ds1_sl1: !!com.alibaba.druid.pool.DruidDataSource
    driverClassName: com.mysql.jdbc.Driver
    url: jdbc:mysql://192.168.8.129:13307/shop
    username: root
    password: root

  ds1_sl2: !!com.alibaba.druid.pool.DruidDataSource
    driverClassName: com.mysql.jdbc.Driver
    url: jdbc:mysql://192.168.8.129:13308/shop
    username: root
    password: root

  ds2: !!com.alibaba.druid.pool.DruidDataSource
    driverClassName: com.mysql.jdbc.Driver
    url: jdbc:mysql://192.168.8.129:13406/shop
    username: root
    password: root

  ds2_sl1: !!com.alibaba.druid.pool.DruidDataSource
    driverClassName: com.mysql.jdbc.Driver
    url: jdbc:mysql://192.168.8.129:13407/shop
    username: root
    password: root

  ds2_sl2: !!com.alibaba.druid.pool.DruidDataSource
    driverClassName: com.mysql.jdbc.Driver
    url: jdbc:mysql://192.168.8.129:13408/shop
    username: root
    password: root

shardingRule:
  tables:
    t_order:
      actualDataNodes: ms_ds${1..2}.t_order${1..3}
      tableStrategy:
        inline:
          shardingColumn: id
          algorithmExpression: t_order${id % 3 + 1}

      databaseStrategy:
        inline:
          shardingColumn: buyer_id
          algorithmExpression: ms_ds${buyer_id % 2 + 1}

      keyGeneratorColumnName: id

  defaultKeyGeneratorClassName: io.shardingsphere.core.keygen.DefaultKeyGenerator

  masterSlaveRules:
    ms_ds1:
      masterDataSourceName: ds1
      slaveDataSourceNames:
        - ds1_sl1
        - ds1_sl2
      loadBalanceAlgorithmType: ROUND_ROBIN

    ms_ds2:
      masterDataSourceName: ds2
      slaveDataSourceNames:
        - ds2_sl1
        - ds2_sl2
      loadBalanceAlgorithmType: ROUND_ROBIN

#props:
#  sql.show: true
```
mysql is 
![image](https://user-images.githubusercontent.com/12459496/50701969-66ef4980-108a-11e9-9330-982c32114e11.png)

and i got the same error
```
Caused by: java.sql.SQLException: Column name pattern can not be NULL or empty.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:545) ~[mysql-connector-java-6.0.6.jar:6.0.6]
```

> ## Bug Report
> **For English only**, other languages will not accept.
> 
> Before report a bug, make sure you have:
> 
> * Searched open and closed [GitHub issues](https://github.com/sharding-sphere/sharding-sphere/issues).
> * Read documentation: [ShardingSphere Doc](http://shardingsphere.io/document/current/en/overview/).
> 
> Please pay attention on issues you submitted, because we maybe need more details.
> If no response **more than 7 days** and we cannot reproduce it on current information, we will **close it**.
> 
> Please answer these questions before submitting your issue. Thanks!
> 
> ### Which version of ShardingSphere did you use?
3.1.0
> ### Which project did you use? Sharding-JDBC or Sharding-Proxy?
 Sharding-JDBC
> ### Expected behavior
run without exception
> ### Actual behavior
Caused by: java.sql.SQLException: Column name pattern can not be NULL or empty.
> ### Reason analyze (If you can)
 if the table in mysql has the same name as actualDataNodes it happens, i dont kown why
> ### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
plz see below
> ### Example codes for reproduce this issue (such as a github link).
plz see below

--- a/sharding-core/src/main/java/io/shardingsphere/core/metadata/table/executor/TableMetaDataLoader.java
+++ b/sharding-core/src/main/java/io/shardingsphere/core/metadata/table/executor/TableMetaDataLoader.java
@@ -133,7 +133,7 @@ public final class TableMetaDataLoader {
     private List<ColumnMetaData> getColumnMetaDataList(final Connection connection, final String catalog, final String actualTableName) throws SQLException {
         List<ColumnMetaData> result = new LinkedList<>();
         Collection<String> primaryKeys = getPrimaryKeys(connection, catalog, actualTableName);
-        try (ResultSet resultSet = connection.getMetaData().getColumns(catalog, null, actualTableName, null)) {
+        try (ResultSet resultSet = connection.getMetaData().getColumns(catalog, null, actualTableName, "%")) {
             while (resultSet.next()) {
                 String columnName = resultSet.getString("COLUMN_NAME");
                 String columnType = resultSet.getString("TYPE_NAME");

